src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs

```diff
@@ -120,25 +128,66 @@ private SafeCreateHandle PreparePoliciesArray(bool checkRevocation)
             return policiesArray;
         }
 
-        private SafeCreateHandle PrepareCertsArray(ICertificatePal cert, X509Certificate2Collection extraStore)
+        private SafeCreateHandle PrepareCertsArray(
+            ICertificatePal cert,
+            X509Certificate2Collection extraStore,
+            X509Certificate2Collection customTrustStore,
+            X509ChainTrustMode trustMode)
         {
-            IntPtr[] ptrs = new IntPtr[1 + (extraStore?.Count ?? 0)];
-            SafeHandle[] safeHandles = new SafeHandle[ptrs.Length];
-
+            int certificateCount = 1 + (extraStore?.Count ?? 0) + (trustMode == X509ChainTrustMode.CustomRootTrust ? customTrustStore?.Count ?? 0 : 0);
+            int certificatesAdded = 1;
```

Comment:
Please move this down to just after the `safeHandles[0] = applePal.CertificateHandle;` line, so that it's clear the added count is being set to 1 because we just stored an element into the array.